### DISCUSSION TITLE
Something went wrong when using ReferenceResidualProblem

### DISCUSSION THREAD

**Discussion by shin-chan-cpu** [idaholab/moose#29626] [Q&A Modules: General]
When using ReferenceResidualProblem, the following error is reported, but I don't know why
```bash
*** ERROR ***
/home/Rookie07/project_202408/bees/HCF/normal_operations.i:9.1:
The following error occurred in the Problem 'MOOSE Problem' of type ReferenceResidualProblem.

Unable to find function
```

**Comment by GiudGiud** [comment]
hello

what does your input file look like?
The Problem is attempting to retrieve a function, seemingly with an empty name, so an object must be requesting that from the problem

**Reply by shin-chan-cpu** [reply]
I searched my input card once, but I didn't find any empty input for function，
If it were really possible, it might happen in the following place
```
[Functions]
  [./power_profile]
    type = PiecewiseLinear  # reads and interpolation an input file containing rod average linear power vs time
    #data_file = history_qbw_new.csv
    data_file = peak_linear_power.csv
    format = columns
  [../]
  [./axial_peaking_factors]
    type = ParsedFunction
    value = 1.0
  [../]
  [./coolant_pressure]
    type = ParsedFunction
    value = 15.5e6
  [../]
  [./q]
    type = CompositeFunction
    functions = 'power_profile axial_peaking_factors'
  [../]
  [./factor]
    type = PowerSurfaceTopFactor
    fuel_diameter = 6.83e-3
    fuel_length =  0.36
  [../]
  [./average_surface_power]
    type = CompositeFunction
    functions = 'power_profile factor'
  [../]
[]
```

**Reply by GiudGiud** [reply]
can you reproduce the error with just this block?

**Reply by shin-chan-cpu** [reply]
This input card is very long, and it's more complicated, I just looked at the part of the function, and it may not be the problem here, but when I change the axial_oeaking_factors option in the functions of the q function to factors, it will report an error that the factor function does not exist, so I wonder if it is a problem here
